src/core/evaluator.js

```diff
@@ -1037,8 +1037,72 @@ var PartialEvaluator = (function PartialEvaluatorClosure() {
         } else {
           textChunk.height += Math.abs(height * scaleCtmX * scaleLineX);
         }
+        addSpaceIfNecessary(textChunk, font);
         return textChunk;
       }
+      function getChunkPosition(chunk, font) {
+        var tx = chunk.transform;
+        var angle = Math.atan2(tx[1], tx[0]);
+        if (font.vertical) {
+          angle += Math.PI / 2;
+        }
+        // Start by calculating the height
+        var fontAscent = Math.sqrt((tx[2] * tx[2]) + (tx[3] * tx[3]));
+        // Then modify to add the ascent
+        if (font.ascent) {
+          fontAscent = font.ascent * fontAscent;
+        } else if (font.descent) {
+          fontAscent = (1 + font.descent) * fontAscent;
+        }
+        return {
+          x : (angle === 0 ? tx[4] : tx[4] + (fontAscent * Math.sin(angle))),
+          y: tx[5]
+        };
+      }
+      function addSpaceIfNecessary(newChunk, font) {
+        // If the new chunk starts with a space, it does not need one.
+        if (newChunk.str[0] === ' ' || newChunk.str[0] === '-') {
+          return;
+        }
+        if (bidiTexts.length === 0) {
+          return;
+        }
+        // If the last chunk ends with a space it does not need one.
+        var lastChunk = bidiTexts[bidiTexts.length - 1];
+        if (lastChunk.str.length === 0) {
+          return;
+        }
+        var lastChar = lastChunk.str[lastChunk.str.length - 1];
+        if (lastChar === ' ' || lastChar === '-') {
+          return;
+        }
+        var lastPosition = getChunkPosition(lastChunk, font);
+        var newPosisition = getChunkPosition(newChunk, font);
+        var yDiff = Math.abs(lastPosition.y  - newPosisition.y);
+        if (yDiff >= lastChunk.height || yDiff >= newChunk.height) {
+          // On different lines, add a space.
+          lastChunk.str += ' ';
+        } else {
+          var wordSpacing = textState.wordSpacing; // Try default wordSpacing.
+          if (wordSpacing === 0) {
+            // Hueristic for wordSpacing
+            wordSpacing = newChunk.width / newChunk.str.length * 0.6;
```

Comment:
Since `0.6` is a heuristic value, it should be replaced with a suitably named constant, placed amongst the other ones at the top of [`getTextContent`](https://github.com/mozilla/pdf.js/blob/master/src/core/evaluator.js#L896-L897).